### ISSUE TITLE
Apple swift enablement

### ISSUE THREAD

**Issue by eldefay** [strongloop/loopback-sdk-ios#13]
I tried to use the SDK in Apple's new swift programming language by bridging to OC. I couldn't call "LBRESTAdapter adapterWithURL". Since the method isn't in "LBRESTAdapter" maybe just adding the .h files of "SLRESTAdapter" would do the trick. I'm just not sure how to do that.

Did anyone try it with swift so far?

**Comment by bajtos** [comment]
@hideya you mentioned that you managed to call iOS client from Swift, could you perhaps advise here?

**Comment by hideya** [comment]
When to use from Swift, we need to pay attention to its automatic remapping of Obj-C factory methods to Swift initializers when importing Obj-C classes (for details please see [this SO post](http://stackoverflow.com/questions/26818432/how-to-call-an-objective-c-class-method-in-swift)).

So, the usage of `LBRESTAdapter adapterWithURL` in Swift world would look like the following:
```swift
let adapter = LBRESTAdapter(URL: NSURL(string: "http://localhost:3000"))
```

I think sample code of iOS SDK usage from Swift should be provided somewhere visible.

**Comment by bajtos** [comment]
> I think sample code of iOS SDK usage from Swift should be provided somewhere visible.

@hideya could you please work with @crandmck to make this happen?

**Comment by crandmck** [comment]
@hideya If you give me a gist or something like that, I will add it to the docs.

**Comment by hideya** [comment]
@crandmck thanks for your contact!  I'll work on a draft and ping you when I finished it.  Please give me a bit of time...

**Comment by bajtos** [comment]
FWIW, a nice doc is available here: http://kevingoedecke.me/2015/12/11/using-loopback-ios-sdk-with-cocoapods-in-swift/

**Comment by hideya** [comment]
And this subsequent doc is very much informative too: http://kevingoedecke.me/2015/12/29/loopback-swift-example-crud-app/

**Comment by hideya** [comment]
I've uploaded a very simple Swift app that shows how to use LB iOS SDK in Swift: https://github.com/hideya/loopback-swift-simple-example
Hope this helps.

**Comment by agfa555** [comment]
This is great!!!!!
Are you going to port the library? I'm having trouble with user authentication. 
What can I use instead of LBPersistedModelAllSuccessBlock? That one can only process an array and I just need an object.
Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Received non-Array'

**Comment by crandmck** [comment]
@hideya Pointed me to https://github.com/hideya/loopback-swift-simple-example.

At first glance it looks pretty good, however, I don't know much about Swift.

I'm happy to addding a link to this in docs, but we'd want to have someone internal at least smoke-test it.  Also, should we consider moving it into an official StrongLoop example, assuming Hideya is willing to contribute it...?

But we should triage this work @superkhau @bajtos.
I don't have any idea of the priority of Swift support.

**Comment by superkhau** [comment]
>Also, should we consider moving it into an official StrongLoop example, assuming Hideya is willing to contribute it...?

I don't know enough swift to maintain it. I mean it's definitely possible, but it'll depend on which one of our resources has the time an expertise to take this on. That and we should it run by @chandadharap before making any decisions on adding any more example repos.

**Comment by crandmck** [comment]
> ...we should it run by @chandadharap before making any decisions on adding any more example repos.

Of course.  It might actually make more sense to add this to an existing example, since it uses the Coffee Shop server app.  But again, not sure how important Swift is.

**Comment by hideya** [comment]
> not sure how important Swift is.

FYI, here are some data points for the discussion:

- [TIOBE Programming Language Index -- Swift #14, Obj-C #18](http://www.tiobe.com/index.php/content/paperinfo/tpci/index.html)
- [PYPL PopularitY of Programming Language -- Swift #9, Obj-C #8](http://pypl.github.io/PYPL.html)
- [Latest Language Ranking: Objective-C Swiftly Falling from Favor](https://adtmag.com/articles/2015/10/14/objective-c-falling.aspx)

It seems that Swift is gaining, and Objective-C is loosing, popularity fairly quickly.  I think Swift support is quite important to gain momentum from the developer community.

**Comment by bajtos** [comment]
> I think Swift support is quite important

+1

I asked a friend of mine who is developing iOS apps, what language does he use these days. The answer was "Swift, unless there is large legacy code base in ObjC".

**Comment by superkhau** [comment]
>I think Swift support is quite important

:+1:

**Comment by agfa555** [comment]
The more I try Swift, the more I like it.
It's trending a lot now and it is OpenSource so expect more developers to use it.
:+1:

**Comment by kgoedecke** [comment]
I'm also currently working with Swift and the LoopBack SDK. That's why I created two projects:
https://github.com/kgoedecke/loopback-swift-crud-example
https://github.com/kgoedecke/loopback-swift-user-example

Hope both help people here!

**Comment by Sequoia** [comment]
@hideya @crandmck @superkhau what is the status here? It sounds like @crandmck is not comfortable adding this to docs due to personal unfamiliarity with Swift. Are you able to add this to the docs @hideya?

**Comment by naomik** [comment]
Swift is represented by birds and birds are cool. +1 for Swift 🐦

**Comment by superkhau** [comment]
I think @jannyHou was working on community projects wiki/docs right? @jannyHou Please chime in. I think we should have a page on loopback.io to support interesting community examples/projects.

**Comment by jannyHou** [comment]
@superkhau Sure I will update that pr asap

**Comment by crandmck** [comment]
@superkhau We have that now!
@jannyHou Community projects docs are here: http://loopback.io/doc/community/index.html

Instructions for adding a page are here: http://loopback.io/doc/en/community/How-to-add-a-project.html  Of course, you can branch the repo instead of forking.  LMK if you have questions.
